adding macos (#447)
Parallels/vagrant-parallels: website/docs/source/docs/boxes/base_macos.html.md

---
page_title: "Creating a macOS Base Box Manually"
sidebar_current: "boxes-macos-base"
---

# Creating a macOS Base Box Manually

Prior to reading this page, please check out the [basics of the Vagrant
box file format](https://www.vagrantup.com/docs/boxes/format.html).

## Pre-requisites

* Install Vagrant, you can find the instructions [here](/vagrant-parallels/docs/installation)

* Start Parallels Desktop and create a new macOS virtual machine as outlined  in [KB 125561](http://kb.parallels.com/125561/).
    <div class="alert alert-info">
    <p>
        While creating the VM, give the username as ‘vagrant’ only.<br />
        If this is a public box you should set the password to ‘vagrant’ as well.
    </p>
    </div>

>   * Enable passwordless sudo for macOS virtual machine
>   * Open Terminal in macOS virtual machine and run:
> 
>      ```bash
>      $ sudo visudo
>      ```
> 
>     Edit the line:
> 
>     ```bash
>     %admin ALL=(ALL) ALL
>     ```
> 
>     To Say:
> 
>     ```bash
>     %admin ALL=(ALL) NOPASSWD: ALL
>     ```
> 
>     Now, you should be able to run sudo without password.

Reviewer: **suggestion:** Let's shorten the section a bit and allow a passwordless sudo for vagrant user only, following the best practices for visudo on macOS:

```suggestion
* Enable passwordless sudo for `vagrant` user on macOS virtual machine:
  * Open Terminal in macOS virtual machine and run:

     ```bash
     $ sudo visudo /private/etc/sudoers.d/vagrant
     ```

    Paste this line there and save the file (`:wq`) :

    ```bash
    vagrant ALL = (ALL) NOPASSWD: ALL
    ```

    Now `vagrant` user is allowed to run `sudo` without a password prompt.
```

Author: implemented

* Install Parallels Tools inside the macOS virtual machine.  
    macOs 13 > **Install Parallels Tools**
  ![install_parallels_extension](/images/install_extension.gif)

      <div class="alert alert-info">
    <p>
        <strong>Note:</strong> Installing the Parallels Tools will require a reboot
    </p>
    </div>

> * Boot macOS virtual machine and enable Remote Login (System Settings > General > Sharing > Enable Remote Login). Don't forget to give “Full disk access to users”, do allow “All users” (in the same settings, press ![info](/images/info_32.png) button)
> ![Parallels Desktop](/images/allow_sharing.gif)
> * Restart macOS virtual machine for the Remote Login to take effect.

Reviewer: **suggestion:** Let's use a better wording and remove the line about the second reboot, since it's not necessary:
```suggestion
* Reboot macOS virtual machine and then enable Remote Login (_System Settings > General > Sharing > Enable Remote Login_). Select “All users” and also enable “Full disk access to users”, as shown on the picture below:
![allow_remote_login](/images/allow_remote_login.gif)
```

Author: implemented

* Copy the vagrant public SSH key to the machine
  
    ```bash
    $ ssh-copy-id -i $(find /opt/vagrant/embedded/gems -type d -name keys)/vagrant vagrant@<virtual_machine_ip>
    ```

>     where:  
> 
>     ```virtual_machine_ip``` \- the virtual machine ip.  

Reviewer: **suggestion:** Let's add the line explaining how to determine the VM IP:
```suggestion
    where `<virtual_machine_ip>` is the current IP of the virtual machine, which could be determined using this command:
    
    ```
    $ prlctl list -f 
    ```
```

## Create Base Box

> * In Parallels directory on your Mac create a folder (something simple like ```VagrantTest```).

Reviewer: **nitpick:** There is no actual need to copy the VM bundle.  I believe, the fastest and easiest way is:
1) Stop the source VM  (by the way, that step should be added in any case, even if you are going to copy the bundle)
2) Unregister the source VM (`prlctl unregister <...>`)
3) Package the box (add metadata, run `tar czvf`), etc.
4) Add the box (`vagrant box add <...>`)
5) Validate the box content - make sure that the VM has `.macvm` name
6) Delete the source macvm bundle and metadata.json (rm -rf <...>), as it's not needed anymore

Benefits: 
- no need for an additional copy - it saves time and the disk space.
- no name collision, since the source VM is unregistered on the early stage.

Author: We will let this method stay as there are a few users that might want to maintain that VM on. we might rethink it in a later stage

## Create a Vagrant folder

We now have our base box created, to create virtual machines from it we will need to create a Vagrant folder structure and a Vagrantfile.

* Create a new folder in the home directory and name it VFTest (In the Finder, you can open your home folder by choosing **Go** > **Home** or by pressing **Shift-Command-H**).

    <div class="alert alert-info">
    <p>
        <strong>Note:</strong> The name of the folder can be different. 'VFTest' is just for example.
    </p>
    </div>
* Download the defautl [Vagrantfile](https://kb.parallels.com/Attachments/kcs-191881/Vagrantfile) and put it in ```VFTest``` folder, or the folder you have created in the previous step.
  ![vagrantfile_folder](/images/vagrantfile_folder.png)

* In Terminal you should go to the directory with Vagrantfile ('VFTest' folder from step 15). Just type:

    ```bash
    $ cd <path to the VFTest folder>
    ```
* You can now start the vagrant box by typing:

    ```bash
    $ vagrant up -provider=parallels
    ```

    This command will start macOS virtual machine (the name of the virtual machine will be different):
    ![macos_vm_started](/images/macos_vm_started.png)

>     Now you are ready to execute Vagrant commands.

Author: implemented

---
page_title: "Creating a macOS Base Box Manually"
sidebar_current: "boxes-macos-base"
---

# Creating a macOS Base Box Manually

Prior to reading this page, please check out the [basics of the Vagrant
box file format](https://www.vagrantup.com/docs/boxes/format.html).

## Pre-requisites

* Download and install [Vagrant](https://developer.hashicorp.com/vagrant/downloads).
* Install Parallels plugin for Vagrant by executing the command below in Terminal:
  
    ```bash
    $ vagrant plugin install vagrant-parallels
    ```

* Start Parallels Desktop and create a new macOS virtual machine as outlined  in [KB 125561](http://kb.parallels.com/125561/).
    <div class="alert alert-info">
    <p>

>         While creating the VM, give the username as ‘vagrant’ only.

Reviewer 2: and if it is a public box, then password = vagrant (recommended according to https://developer.hashicorp.com/vagrant/docs/boxes/base#)

Author: Added that mention

    </p>
    </div>

  * Enable passwordless sudo for macOS virtual machine
  * Open Terminal in macOS virtual machine and run:

     ```bash
     $ sudo visudo
     ```

    Edit the line:

    ```bash
    %admin ALL=(ALL) ALL
    ```

    To Say:

    ``` bash
    %admin ALL=(ALL) NOPASSWD: ALL
    ```

>     Now, you should now be able to run sudo without password.

Reviewer 2: One extra "now".

Author: removed

> * Install Parallels Tools inside the macOS virtual machine.

Reviewer 2: a GIF or a link on how to do it would be good. and also that it is essential for vagrant to work and enables copy/paste, shared folders? , prlctl exec

Author: Added a gif and aa better description

* Boot macOS virtual machine and enable Remote Login (System Settings > General > Sharing > Enable Remote Login). Don't forget to give “Full disk access to users”, do allow “All users” (in the same settings, press ![info](/images/info_32.png) button)
![Parallels Desktop](/images/allow_sharing.gif)
* Restart macOS virtual machine for the Remote Login to take effect.

## SSH Keys

> We will need to create a new SSH keypair for Vagrant to use to communicate

Reviewer 2: or use public keypair from https://github.com/hashicorp/vagrant/tree/main/keys

Author: done

* Copy the initially created bundle of your macOS virtual machine (it has the **.macvm** extension) to the newly created folder (```VagrantTest```).
    
    <div class="alert alert-info">
    <p>
        <strong>Optional:</strong> You can rename macOS virtual machine with a unique name so that you won't get any name collision issues.
    </p>
    </div>

  * Run Terminal on the host side and execute the following command:
  
    ``` bash
    $ prlctl set "macOS 13" --name "macOS_13"
    ```

    Where the name 'macOS_13' is your virtual machine's name.

    <div class="alert alert-info">
    <p>
        <strong>Note:</strong> Renaming the virtual machine is one of the solutions to avoid name collision. You can also remove macOS virtual machine from Control Center (Parallels icon > Control Center > Right-click on macOS 13 virtual machine > Remove 'macOS 13' > <strong>Keep Files</strong>).
    </p>
    </div>

* Download [Vagrantfile](https://kb.parallels.com/Attachments/kcs-191881/Vagrantfile) and put it in the VagrantTest folder.
* Download [metadata.json](https://kb.parallels.com/Attachments/kcs-191881/metadata.json) file and put it in the VagrantTest folder.

> * Copy your **private** key from ~/.ssh/ directory in VagrantTest directory and rename it to '**vagrant\_private\_key**':

Reviewer 2: no need to do this if using the public pair.

Author: added an alert that if they choose the public vagrant key they would not need to do that part

---
page_title: "Creating a macOS Base Box Manually"
sidebar_current: "boxes-macos-base"
---

# Creating a macOS Base Box Manually

Prior to reading this page, please check out the [basics of the Vagrant
box file format](https://www.vagrantup.com/docs/boxes/format.html).

## Pre-requisites

* Install Vagrant, you can find the instructions [here](/vagrant-parallels/docs/installation)

* Start Parallels Desktop and create a new macOS virtual machine as outlined  in [KB 125561](http://kb.parallels.com/125561/).
    <div class="alert alert-info">
    <p>
        While creating the VM, give the username as ‘vagrant’ only.<br />
        If this is a public box you should set the password to ‘vagrant’ as well.
    </p>
    </div>

  * Enable passwordless sudo for macOS virtual machine
  * Open Terminal in macOS virtual machine and run:

     ```bash
     $ sudo visudo
     ```

    Edit the line:

    ```bash
    %admin ALL=(ALL) ALL
    ```

    To Say:

>     ``` bash

Reviewer 3: ```suggestion
    ```bash
```

Author: fixed
